sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ResponseUtils.java

```diff
@@ -24,20 +25,30 @@ static Mono<StoreResponse> toStoreResponse(HttpResponse httpClientResponse, Stri
             // transforms to Mono<StoreResponse>
             int size = 0;
             if (byteBufContent == null || (size = byteBufContent.readableBytes()) == 0) {
+                try {
+                    return new StoreResponse(
+                        endpoint,
+                        httpClientResponse.statusCode(),
+                        HttpUtils.unescape(httpResponseHeaders.toMap()),
+                        null,
+                        0,
+                        null);
+                } catch (Exception e) {
+                    throw reactor.core.Exceptions.propagate(e);
+                }
+            }
+
+            try {
                 return new StoreResponse(
                     endpoint,
                     httpClientResponse.statusCode(),
                     HttpUtils.unescape(httpResponseHeaders.toMap()),
-                    null,
-                    0);
+                    new ByteBufInputStream(byteBufContent, true),
+                    size,
+                    null);
+            } catch (Exception e) {
+                throw reactor.core.Exceptions.propagate(e);
```

Comment:
This is catching all exceptions including checked and unchecked, is that what we want here?
Ideally we should just catch checked exceptions here and let unchecked exceptions take their own path as they used to earlier to keep the original behavior unless there was a problem with the original behavior.

Reply:
This is primarily due to the interceptor logic (won't be present in public releases).